File: Fucapi/2013_1/Analise/Trabalho 1/gerarNumeros.go
package main

import (
	"fmt"
	"os"
	"math"
	"strconv"
	"math/big"
	"crypto/rand"
)

var (
	Max = big.NewInt(int64(math.MaxUint16))
	Saida = "numeros"
	Quant = 100000
)

func main() {
	args := os.Args;
	
	switch (len(args)) {
	case 3:
		Saida = args[2]
		fallthrough
	case 2:
		Quant, _ =  strconv.Atoi(args[1])
	}
	
	arquivo, err := os.Create(Saida)
	
	if err != nil {
		fmt.Println("Erro ao abrir arquivo:", err)
		return
	}
	defer arquivo.Close()
	
	for indice := 0; indice < Quant; indice++ {
		num, err := rand.Int(rand.Reader, Max)
		if err != nil {
			fmt.Println("Erro ao gerar número:", err)
			return
		}
		
		if num.Int64() > int64(math.MaxUint16) {
			fmt.Println("Gerou número acima do limite!")
			return
		}
		fmt.Fprintln(arquivo, num.Int64())
	}
}

File: Fucapi/2013_1/Analise/Trabalho 1/gerarNumeros.c
#include <stdlib.h>
#include <stdio.h>
#include <inttypes.h>
#include <time.h>

int main(int argc, char* argv[]) {
	FILE *arquivo;
	char *saida;
	int quant;
	
	saida = "numeros";
	quant = 10000;
	
	switch (argc) {
		case 3:
			saida = argv[2];
		case 2:
			quant = atoi(argv[1]);
			break;
	}
	
	arquivo = fopen(saida, "w");
	if (arquivo == NULL) {
		printf("Erro ao criar saida!\n");
		return 0;
	}
	
	srand(time(NULL));
	
	for (int indice = 0; indice < quant; indice++) {
		fprintf(arquivo, "%d\n", rand() % UINT16_MAX);
	}
	
	fclose(arquivo);
	
	return 0;
}
